pkg/cmd/cli/debug/debug.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2017 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	_ "embed"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	crashdCmd "github.com/vmware-tanzu/crash-diagnostics/cmd"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/vmware-tanzu/velero/pkg/client"
+	"github.com/vmware-tanzu/velero/pkg/cmd"
+)
+
+//go:embed cshd-scripts/velero.cshd
+var scriptBytes []byte
+
+type option struct {
+	// workdir for crashd will be $baseDir/velero-debug
+	baseDir string
+	// the namespace where velero server is installed
+	namespace string
+	// the absolute path for the log bundle to be generated
+	outputPath string
+	// the absolute path for the kubeconfig file that will be read by crashd for calling K8S API
+	kubeconfigPath string
+	// optional, the name of the backup resource whose log will be packaged into the debug bundle
+	backup string
+	// optional, the name of the restore resource whose log will be packaged into the debug bundle
+	restore string
+}
+
+func (o *option) bindFlags(flags *pflag.FlagSet) {
+	flags.StringVar(&o.outputPath, "output", "", "The path of the bundle tarball, by default it's $HOME/bundle.tar.gz. Optional")
```

Comment:
When I ran it, it output into my current directory, not $HOME.  I think that's actually better behavior, so let's change the document here.

Reply:
Let's make the default name generated more descriptive and unique.  I'm thinking `"velero-debug-<YYYY>-<MM>-<DD>-<HH>-<MM>-<SS>.tar.gz"`  Or we could use the crashd # and that would be consistent.

Reply:
Also, when I untar it, it goes into tmp/crashdXXXX.  Can we not put the tmp part of the path into the tar file?

Reply:
Fixed according to the 3 comments.

Reply:
If I run this without the --output flag, it still states that it created a `bundle.tar.gz` file (set in [the script](https://github.com/vmware-tanzu/velero/pull/4022/files#diff-759a9f8572722bad288275ce971213be28113874907c5e273565b03c624b9d65R13), but I don't know where it was written to. I don't see any file in the current directory, my home directory, or /tmp.

Reply:
> but I don't know where it was written to. I don't see any file in the current directory, my home directory, or /tmp.

I see it is actually in the /tmp/crashd* directory. I think we need to set the `outputPath` field correctly before passing it in so it is written to the specified location, or in the current directory.

Reply:
I did make it generate the filename according to Dave's comment, seems the change was lost after rebasing ... I'll add it back, sorry.